# ğŸ§¬ DGS Final Project â€“ Monogenic Diabetes Variant Analysis

Monogenic diabetes is a rare form of diabetes that results from a mutation in a single gene. This genetic change can affect the way the body produces or uses insulin, leading to high blood sugar levels. Unlike Type 1 or Type 2 diabetes, monogenic diabetes often appears earlier in life and sometimes can be managed with targeted treatments, such as specific oral medications rather than insulin. Understanding the genetic basis of monogenic diabetes is key to accurate diagnosis and personalized care.

## Aim of the Project
The following project evaluates the clinical utility of the Delfos platform by comparing its ability to identify Monogenic Diabetes-related genetic variants against ClinGen, the current gold standard.
The project was divided into two main parts:

1. **Knowledge Base Integration** using Hermes, Ulises, Delfos, and Sibila.
2. **Variant Comparison** using SQL-based evaluation of ClinGen vs DELFOS data.

---

## ğŸ“ Project Structure

```
DGS_FINAL_PROJECT/
â”‚
â”œâ”€â”€ db/                                       # Database-related files
â”‚   â””â”€â”€ variants.db                           # SQLite database storing genetic variants
â”‚
â”œâ”€â”€ data/                                     # Input data files (VCF format)
â”‚   â”œâ”€â”€ VCF_clingen.vcf                       # VCF file from ClinGen
â”‚   â”œâ”€â”€ VCF_ulises.vcf                        # VCF file generated from Ulises module
â”‚   â””â”€â”€ VCF_ulises-p2.vcf                     # VCF file generated from Ulises module (all besides rejected)
â”‚
â”œâ”€â”€ scripts/                                  # All Python scripts used in the project
â”‚   â”œâ”€â”€ hermesScript.R                        # Retrieves data from Hermes
â”‚   â”œâ”€â”€ ulises_to_vcf.py                      # Converts Ulises data to VCF (provided by instructors)
â”‚   â”œâ”€â”€ load_data.py                          # SQLite database setup
â”‚   â”œâ”€â”€ insert_vcfs.py                        # Loading VCF to database
â”‚   â””â”€â”€ metrics.py                            # Comparison of variants from both datasets
â”‚
â”œâ”€â”€ results/                                  # Output files and results from modules
â”‚   â”œâ”€â”€ hermes/                               # Results generated by Hermes
â”‚   â”‚   â”œâ”€â”€ hermes_data.json
â”‚   â”‚   â””â”€â”€ relevant_genes.json
â”‚   â”œâ”€â”€ ulises/                               # Processed data from Ulises
â”‚   â”‚   â”œâ”€â”€ data-processed.json
â”‚   â”‚   â””â”€â”€ data-processed-p2.json
â”‚   â”œâ”€â”€ concordance_summary.csv               # Comparison results (number of variants for each interpretation match)
â”‚   â”œâ”€â”€ interpretation_concordance_analysis   # Detailed comparison for each variant
â”‚   â””â”€â”€ statistical_parameters_summary        # Key metrics (total variants analyzed)
â”‚
â”œâ”€â”€ docker-compose.yaml                       # Docker Compose configuration for project setup
â”œâ”€â”€ LICENSE                                   # License information
â”œâ”€â”€ README.md                                 # Project overview and setup instructions
â””â”€â”€ .gitignore                                # Git ignore file for excluding untracked files
```

## ğŸ‘¨ğŸ»â€ğŸ« Presentation
Click [here](https://www.canva.com/design/DAGjSsQyAbA/1Lu7jkZztlKkHk9UyDtXvA/edit?utm_content=DAGjSsQyAbA&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton) to access the presentation.


## ğŸ‘¥ Authors

- **Marcin Grzelak**  
  ğŸ“§ [mgrzela@upv.edu.es](mailto:mgrzela@upv.edu.es)

- **Zuzanna PajÄ…k**  
  ğŸ“§ [zpajak@upv.edu.es](mailto:zpajak@upv.edu.es)
